telegram-history-dump.rb

```diff
@@ -165,10 +175,20 @@ def process_media(dialog, msg)
     end
     filename = case
       when $config['copy_media']
-        filename = File.basename(response['result'])
-        destination = File.join(get_media_dir(dialog), fix_media_ext(filename))
-        FileUtils.cp(response['result'], destination)
-        destination
+        if response.nil?
+          $log.error('Missing media for message id %s' % msg['id'])
+          "missing_media_%s" % msg['id']
+        else
+          begin
+            filename = File.basename(response['result'])
+            destination = File.join(get_media_dir(dialog), fix_media_ext(filename))
+            FileUtils.cp(response['result'], destination)
+            destination
+          rescue StandardError => e
+            $log.error('Missing media for message id %s' % msg['id'])
+            "missing_media_%s" % msg['id']
+          end
+        end
       else
```

Comment:
I think it's a good addition to check for an invalid response. Perhaps an additional type check would make it even more solid, i.e. `if response.nil? || !response.is_a?(Hash)`. This check could even be the first condition of the `case` block and just return `nil` as filename.

It's also a good idea to indicate in the output that the file download failed (with current master the file property is not set at all if it is not valid), but I'm not so sure that replacing the filename with `missing_media_%s` is a good approach. That's because it's a valid string, so people will expect it to be a valid filename when processing the output. I'd be in favor of just setting `msg['media']['file']` to `nil` or `false`, this will translate to `null`/`false` in the JSON. It's less explicit than your version but easier to process with code.

I'm in doubt about the begin-rescue block around the File and FileUtils calls. While there is an argument for continuing the dump whenever possible, if one of these functions fails it is probably not the fault of telegram-cli but rather the result of an error in the script configuration or a problem with the host system (e.g. filesystem permissions). Because of this I believe it may be better to just let the exception through when this part fails. I can think of one exception: if the source file (the media file in the telegram-cli directory) does not exist it may be the fault of telegram-cli, or caused by running telegram-cli remotely on a different machine. In this situation it would be good to rescue. I believe this can be done by rescuing a specialized error class, I can do this if you want.